src/i_winmusic.c

```diff
@@ -52,8 +52,22 @@ static byte xg_system_on[] = {
     0xF0, 0x43, 0x10, 0x4C, 0x00, 0x00, 0x7E, 0x00, 0xF7
 };
 
-static int timediv;
-static int tempo;
+static byte master_volume_msg[] = {
+    0xF0, 0x7F, 0x7F, 0x04, 0x01, 0x7F, 0x7F, 0xF7
+};
+
+#define DEFAULT_MASTER_VOLUME 16383
+static unsigned int master_volume = DEFAULT_MASTER_VOLUME;
+static int last_volume = -1;
+static float volume_factor = 1.0f;
+static boolean update_volume = false;
+
+#define MAX_SYSEX_LENGTH 1024
```

Comment:
Is this limit too small? For example, loopMIDI sets a limit of 256 KB by default.

Reply:
> My understanding is that master volume SysEx is part of the GM standard and should be supported by everyone, right? At least all the softsynths I've tried work.

Yes, every MIDI device should be able to receive master volume.

Reply:
I don't have a good feel for what the limit should be. PrBoom+ uses 1024 bytes but that may have just been a placeholder for the old SysEx code that didn't work. The longest SysEx I've seen in a song is 26 bytes (voice reserve tweaks for Roland hardware). The MIDI file specs say that SysEx length is a "variable length quantity" which means the largest number it can be is  0x0FFFFFFF (268 MB!). I'm open to suggestions. 256KB is fine by me though.

Reply:
My suggestion is to remove the limit, I think winmm/driver will check it for us anyway.

Reply:
Fixed by https://github.com/fabiangreffrath/woof/pull/822/commits/78b91c597c2ec78031e586c9ff7ec4c51c20ed5b